addons/mail/static/src/discuss/core/web/messaging_menu_patch.xml

```diff
@@ -3,24 +3,20 @@
     <t t-inherit="mail.MessagingMenu.content" t-inherit-mode="extension">
         <xpath expr="//div[hasclass('o-mail-MessagingMenu-header')]" position="inside">
             <div class="flex-grow-1"/>
-            <t t-if="!store.discuss.isActive or ui.isSmall and store.discuss.activeTab !== 'mailbox'">
-                <t t-if="env.inDiscussApp and ['chat', 'channel'].includes(store.discuss.activeTab)">
-                    <div t-if="state.addingChat or state.addingChannel" class="p-4 border-bottom bg-light" t-ref="selector">
-                        <t t-if="state.addingChat" t-set="category" t-value="store.discuss.chats"/>
-                        <t t-if="state.addingChannel" t-set="category" t-value="store.discuss.channels"/>
-                        <ChannelSelector category="category" autofocus="true"/>
-                    </div>
-                </t>
-                <button t-if="!env.inDiscussApp or (env.inDiscussApp and store.discuss.activeTab === 'chat' and !state.addingChat)" t-att-class="env.inDiscussApp ? 'w-100 p-2 btn btn-secondary border-bottom bg-light' : 'btn btn-link'" t-on-click.stop="onClickNewMessage">
-                    <t t-if="env.inDiscussApp">Start a conversation</t>
-                    <t t-else="">New Message</t>
-                </button>
-                <div t-if="env.inDiscussApp and store.discuss.activeTab === 'channel' and !state.addingChannel">
-                    <button class="w-100 p-2 btn btn-secondary border-bottom bg-light" t-on-click.stop="() => this.state.addingChannel = true">
-                        New Channel
-                    </button>
+            <t t-if="ui.isSmall and (store.discuss.activeTab !== 'main' or !env.inDiscussApp)">
+                <div t-if="state.addingChat or state.addingChannel" class="w-100 p-1 border-bottom bg-light" t-ref="selector">
+                    <t t-if="state.addingChat" t-set="category" t-value="store.discuss.chats"/>
+                    <t t-if="state.addingChannel" t-set="category" t-value="store.discuss.channels"/>
+                    <ChannelSelector category="category" autofocus="true"/>
                 </div>
             </t>
+            <button t-if="!ui.isSmall or (ui.isSmall and store.discuss.activeTab !== 'channel' and !state.addingChat and !(env.inDiscussApp and store.discuss.activeTab === 'main'))" t-att-class="ui.isSmall ? 'w-100 p-2 btn btn-secondary border-bottom bg-light' : 'btn btn-link p-2'" t-on-click.stop="onClickNewMessage">
+                <t t-if="ui.isSmall">Start a conversation</t>
```

Comment:
It looks off when there's no button on the livechat button, because the initial content is move up.

This button and the mailbox selection have the same height, ensuring the content part doesn't move up/down when changing tab.

One technique could be to keep same area as if the button is there but it's not (e.g. with `visibility: hidden`), but I think it's a better UX/UI to always have a button available.

Currently there's "New message"/"Start a conversation"/"New channel". The former 2 are the same except for label. It would be nice for follow-up work in master to have a single button for both "Start a conversation" and "New channel", as it feels best to have a single button that's always present that always do the same thing. There's already many changes of screens with the tabs and mailbox selection, the contextual buttons is too much and avoidable.

Reply:
But starting a livechat conversation (from that menu in particular) makes no sense. And if it's creating a DM instead, it makes even less sense in the livechat menu.

Reply:
Probably the best in short-term is to remove the button in livechat tab.
This part of UI in mobile has ongoing tasks that will require more thought, e.g. having "Start a meeting" in the mobile screen. One idea was to put it in this header, but there's likely other and better options